### ISSUE TITLE
Mocking the Client Service 

### ISSUE THREAD

**Issue by kaungsoe** [grpc/grpc-swift#780]
### Question Checklist

- [x] Updated grpc-swift to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/grpc/grpc-swift/blob/master/CONTRIBUTING.md)
- [x] I read [README](https://github.com/grpc/grpc-swift/blob/master/README.md)
- [x] I searched for [existing GitHub issues](https://github.com/grpc/grpc-swift/issues)

### Question Subject

Mocking the client service

#### Question Description

Here is the client service from example.
```
public protocol Routeguide_RouteGuideClientProtocol {
  func getFeature(_ request: Routeguide_Point, callOptions: CallOptions?) -> UnaryCall<Routeguide_Point, Routeguide_Feature>
  func listFeatures(_ request: Routeguide_Rectangle, callOptions: CallOptions?, handler: @escaping (Routeguide_Feature) -> Void) -> ServerStreamingCall<Routeguide_Rectangle, Routeguide_Feature>
  func recordRoute(callOptions: CallOptions?) -> ClientStreamingCall<Routeguide_Point, Routeguide_RouteSummary>
  func routeChat(callOptions: CallOptions?, handler: @escaping (Routeguide_RouteNote) -> Void) -> BidirectionalStreamingCall<Routeguide_RouteNote, Routeguide_RouteNote>
}
```
What I want to do is, create mock object for this service. 
```
class Routeguide_RouteGuideClientMock: Routeguide_RouteGuideClientProtocol {
    
    func getFeature(_ request: Routeguide_Point, callOptions: CallOptions?) -> UnaryCall<Routeguide_Point, Routeguide_Feature> {
        // return mock object directely
    }
    
    func listFeatures(_ request: Routeguide_Rectangle, callOptions: CallOptions?, handler: @escaping (Routeguide_Feature) -> Void) -> ServerStreamingCall<Routeguide_Rectangle, Routeguide_Feature> {
        
    }
    
    func recordRoute(callOptions: CallOptions?) -> ClientStreamingCall<Routeguide_Point, Routeguide_RouteSummary> {
        
    }
    
    func routeChat(callOptions: CallOptions?, handler: @escaping (Routeguide_RouteNote) -> Void) -> BidirectionalStreamingCall<Routeguide_RouteNote, Routeguide_RouteNote> {
        
    }
}
```
But there is no public initialiser. Is there any way I can accomplish it. Thanks.

**Comment by MrMage** [comment]
On the old `cgrpc` branch, we actually had special "test stub" implementations of these classes to make mocking even easier. Unfortunately, we have not gotten around yet to prioritizing this for the new NIO branch. @glbrntt do you think this is something we could prioritize?

**Comment by glbrntt** [comment]
We should prioritise this, yes. I don't think I'll have time to get to it this week though (next week hopefully). 

Some initial thoughts:
- We can probably do this without having to construct a channel at all; we'll still need an event loop (`EmbeddedEventLoop` might be okay here)
- Some of the features available on the call object won't be available: the http/2 stream channel future and RPC cancellation to name some; maybe that's not an issue, however.

**Comment by MrMage** [comment]
FYI, #800 should solve this when done.

**Comment by glbrntt** [comment]
Resolved by #855,  #864, and #870
